test/common/upstream/health_check_fuzz.proto

```diff
@@ -13,8 +13,45 @@ message HttpRespond { //TODO: Split this across headers, body, and trailers?
   string status = 2 [(validate.rules).string = {pattern: "[0-9]+$*", min_len: 1}];
 }
 
+message TcpRespond {
+  bytes data = 1;
+}
+
+message GrpcRespondHeaders {
+  test.fuzz.Headers headers = 1;
+  //https://github.com/nodejs/http-parser/blob/master/http_parser.c#L852
+  string status = 2 [(validate.rules).string = {pattern: "[0-9]+$*", min_len: 1}];
+}
+
+enum ServingStatus {
+  UNKNOWN = 0;
+  SERVING = 1;
+  NOT_SERVING = 2;
+  SERVICE_UNKNOWN = 3; // Used only by the Watch method.
+}
+
+message GrpcRespondBytes {
+  oneof grpc_respond_bytes_selector {
+    //Structured response, which will get converted to raw bytes
+    ServingStatus status = 1;
+    bytes data = 2; //Raw bytes
+  }
+}
+
+message GrpcRespondTrailers {
+  test.fuzz.Headers trailers = 1;
+}
+
+message GrpcRespond {
+  GrpcRespondHeaders grpc_respond_headers = 1;
+  GrpcRespondBytes grpc_respond_bytes = 2;
+  GrpcRespondTrailers grpc_respond_trailers = 3;
```

Comment:
Left both as a message. Left because added fuzzed status in grpc_respond_headers, and wanted to keep it consistent for grpcrespondtrailers. Also, if I decide to fuzz different grpc-statuses, this will allow me to do so. However, if you feel as though itd be best to make trailers test.fuzz.Headers, let me know!